lib/widgets/action_sheet.dart

```diff
@@ -317,3 +319,51 @@ class CopyButton extends MessageActionSheetMenuItemButton {
       data: ClipboardData(text: rawContent));
   };
 }
+
+class StarButton extends MessageActionSheetMenuItemButton {
+  StarButton({
+    super.key,
+    required super.message,
+    required super.messageListContext,
+  });
+
+  @override get icon => ZulipIcons.star;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return message.flags.contains(MessageFlag.starred)
+      ? zulipLocalizations.actionSheetOptionUnstarMessage
+      : zulipLocalizations.actionSheetOptionStarMessage;
+  }
+
+  @override get onPressed => (BuildContext context) async {
+    Navigator.of(context).pop();
+    final zulipLocalizations = ZulipLocalizations.of(messageListContext);
+    final op = message.flags.contains(MessageFlag.starred)
+      ? UpdateMessageFlagsOp.remove
+      : UpdateMessageFlagsOp.add;
+
+    try {
+      final connection = PerAccountStoreWidget.of(messageListContext).connection;
+      await updateMessageFlags(connection, messages: [message.id],
+        op: op, flag: MessageFlag.starred);
+    } catch (e) {
+      if (!messageListContext.mounted) return;
+      String? errorMessage;
+
+      switch (e) {
```

Comment:
nit: this `errorMessage` variable doesn't have anything to do with the if-unmounted-return, but is related to the switch because that's what can set it. So it's weird to have it attached to the former and separated from the latter.

```suggestion
      if (!messageListContext.mounted) return;

      String? errorMessage;
      switch (e) {
```